drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -54,6 +55,7 @@ config ETH_STM32_HAL_RANDOM_MAC
 	default y
```

Comment:
If this is deprecated, should select DEPRECATED and also not be default y

Reply:
And add ` (DEPRECATED)` or similar to end of the text too e.g. `bool "Random MAC address (DEPRECATED)"`

Reply:
Ok will update